Move Westwood file formats to Mods.Cnc/D2k

reaperrr opened the pull request:
This PR moves all remaining proprietary WW file formats to the Westwood-specific mod code, except (for now) ShpTS, due to its non-proprietary RLEZeros encoding and good community tool support.

The idea is to make it unnecessary for potential downstream commercial games to rip stuff out of Mods.Common out of legal concerns.

Note: The VQA changes are actually still somewhat WIP, as playing FMV in radar is untested (potentially broken) and playing in Asset Browser is broken (not sure why).
Help would be appreciated.

Closes #12775.

Mailaender commented on OpenRA.Mods.Common/Scripting/Media.cs:
```diff
@@ -22,7 +22,7 @@ public static class Media
 		{
 			var playerRoot = Game.OpenWindow(w, "FMVPLAYER");
-			var player = playerRoot.Get<VqaPlayerWidget>("PLAYER");
+			var player = playerRoot.Get<VideoPlayerWidget>("PLAYER");
```
I don't understand the reason for this refactor. You create a new "generic" `VideoPlayerWidget`, but the only format supported is .vqa anyway. It may be simpler to move everything touching video support to Mods.Cnc to avoid common depending on cnc, which seems to be your initial motivation.

GraionDilach replied:
This is running ahead a bit and is setup to suit with, say, an MPEG-2 decoder at hand, whose mp2 support can stay within Mods.Common.

reaperrr replied:
Yeah, the idea is to avoid having to move back Media and MediaGlobal once we support some generic, non-proprietary video format.

Mailaender replied:
What I mean is that you seem to be handling inheritance the wrong way. You are referencing the base/placeholder class here. It doesn't automatically resolve as the correct `VqaPlayerWidget` automagically. You only get the generic stub, so I am not surprised it doesn't work as intended.

My suggestion would be to have a fully functional `VideoPlayerWidget` and let it decide for the file format by looking for interfaces that check for magic bytes or file endings and then get back a `VqaReader` stream. Or if you want to keep `VqaPlayerWidget` then let the media routine here pick the correct widget based on what file it is.

Mailaender replied:
See how https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/Sound/Sound.cs handles first the check if it's the turn of the audio loader and then returns a stream.

reaperrr committed:
Move AUD, VQA and LCWCompression to Mods.Cnc

These are the last proprietary formats
in Mods.Common.
We're leaving ShpTS in Mods.Common for now
for its wide-spread tool support and
non-proprietary RLEZerosCompression.

reaperrr commented:
Closing this for now since I don't know how long it'll take to have something PR-able, and then it would probably be cleaner to file a new PR with the VQA stuff anyway.
